src/tools/kata-ctl/src/main.rs

```diff
@@ -3,122 +3,35 @@
-    let version = version::get();
-
-    let app = App::new(name)
-        .version(&*version)
-        .about(ABOUT_TEXT)
-        .long_about(DESCRIPTION_TEXT)
-        .subcommand(
-            SubCommand::with_name("check")
-                .about("tests if system can run Kata Containers")
-                .arg(
-                    Arg::with_name("no-network-checks")
-                        .long("no-network-checks")
-                        .help("run check with no network checks")
-                        .takes_value(false),
-                ),
-        )
-        .subcommand(
-            SubCommand::with_name("direct-volume")
-                .about("directly assign a volume to Kata Containers to manage"),
-        )
-        .subcommand(SubCommand::with_name("env").about("display settings. Default to TOML"))
-        .subcommand(SubCommand::with_name("exec").about("enter into guest by debug console"))
-        .subcommand(SubCommand::with_name("factory").about("manage vm factory"))
-        .subcommand(
-            SubCommand::with_name("help").about("shows a list of commands or help for one command"),
-        )
-        .subcommand(SubCommand::with_name("iptables").about(""))
-        .subcommand(
-            SubCommand::with_name("metrics")
-                .about("gather metrics associated with infrastructure used to run a sandbox"),
-        )
-        .subcommand(SubCommand::with_name("version").about("display version details"));
-
-    let args = app.get_matches();
-
-    let subcmd = args
-        .subcommand_name()
-        .ok_or_else(|| anyhow!("need sub-command"))?;
-
-    match subcmd {
-        "check" => {
-            match run_checks(args) {
-                Ok(_result) => println!("check may not be fully implemented"),
-                Err(err) => println!("{}", err),
-            }
-            Ok(())
-        }
-        "direct-volume" => {
-            unimplemented!("Not implemented");
-        }
-        "env" => {
-            unimplemented!("Not implemented");
-        }
-        "exec" => {
-            unimplemented!("Not implemented");
-        }
-        "factory" => {
-            unimplemented!("Not implemented");
-        }
-        "help" => {
-            unimplemented!("Not implemented");
-        }
-        "iptables" => {
-            unimplemented!("Not implemented");
-        }
-        "metrics" => {
-            unimplemented!("Not implemented");
-        }
-        "version" => {
-            println!("{} version {} (type: rust)", NAME, version);
-            Ok(())
-        }
-        _ => return Err(anyhow!(format!("invalid sub-command: {:?}", subcmd))),
-    }
-}
+use ops::check_ops::handle_check_command;
+use ops::check_ops::handle_check_volume;
+use ops::check_ops::handle_env;
+use ops::check_ops::handle_exec;
+use ops::check_ops::handle_factory;
+use ops::check_ops::handle_iptables;
+use ops::check_ops::handle_metrics;
+use ops::check_ops::handle_version;
 
 fn main() {
```

Comment:
Could you either change `main()` to return a `Result`:

```suggestion
fn main() -> Result<()> {
```

... or, re-instate the `real_main()` function and have it call the handlers as you are doing below.

tbh, I'd prefer to keep the `real_main()` and the reason is subtle: that `eprintln!("ERROR: {:#?}", e);` in that function is a bit "magic" in that the `{:#?}` format will dtrt if we start using [`anyhow::Context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#example), which we will probably want. I don't _think_ we'll get the same behaviour if we just return the `Result` from `main()` (although admittedly I haven't tested that).

Reply:
I also would prefer to keep real_main()

Reply:
thanks @jodh-intel @cmaf , fixed in the next PR attempt